sipvicious/svmap.py

```diff
@@ -423,92 +450,101 @@ def main():
+
     if options.save is not None:
         if options.resume is None:
             exportpath = os.path.join(os.path.expanduser('~'),'.sipvicious',__prog__,options.save)
             if os.path.exists(exportpath):
-                logging.warning('we found a previous scan with the same name. Please choose a new session name')
-                exit(1)
+                parser.error('we found a previous scan with the same name. Please choose a new session name', 20)
+
             logging.debug('creating an export location %s' % exportpath)
             try:
                 os.makedirs(exportpath,mode=0o700)
             except OSError:
-                logging.critical('could not create the export location %s' % exportpath)
-                exit(1)
+                parser.error('could not create the export location %s' % exportpath, 20)
+
             optionsdst = os.path.join(exportpath,'options.pkl')
             logging.debug('saving options to %s' % optionsdst)
             pickle.dump([options,args],open(optionsdst,'wb+'))
+
     try:
         options.extension
     except AttributeError:
         options.extension = None
+
     if options.autogetip:
         tmpsocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         tmpsocket.connect(("msn.com",80))
         options.externalip=tmpsocket.getsockname()[0]
         tmpsocket.close()
+
     sipvicious = DrinkOrSip(
-                    scaniter,
-                    selecttime=options.selecttime,
-                    compact=options.enablecompact,
-                    localport=options.localport,
-                    externalip=options.externalip,
-                    bindingip=options.bindingip,
-                    sessionpath=exportpath,
-                    extension=options.extension,
-                    printdebug=options.printdebug,
-                    first=options.first,
-                    fromname=options.fromname,
-                    ipv6=options.ipv6,
-                    )
+        scaniter,
+        selecttime=options.selecttime,
+        compact=options.enablecompact,
+        localport=options.localport,
+        externalip=options.externalip,
+        bindingip=options.bindingip,
+        sessionpath=exportpath,
+        extension=options.extension,
+        printdebug=options.printdebug,
+        first=options.first,
+        fromname=options.fromname,
+        ipv6=options.ipv6,
+    )
     start_time = datetime.now()
-    logging.info( "start your engines" )
+    logging.info("start your engines")
+
     try:
-        try:
-            sipvicious.start()
-        except AssertionError as err:
-            logging.critical(err)
-            exit(1)
+        sipvicious.start()
         if exportpath is not None:
             open(os.path.join(exportpath,'closed'),'w').close()
+
     except KeyboardInterrupt:
         logging.warning( 'caught your control^c - quiting' )
+        __exitcode__ = resolveexitcode(20, __exitcode__)
```

Comment:
same here - don't set exit code because of keyboard interrupt

Reply:
taken care in 3d909a7.